lms/djangoapps/course_api/blocks/api.py

```diff
@@ -43,24 +45,21 @@ def get_blocks(
         return_type (string): Possible values are 'dict' or 'list'. Indicates
             the format for returning the blocks.
     """
-    # construct BlocksAPITransformer
-    blocks_api_transformer = BlocksAPITransformer(
-        block_counts,
-        student_view_data,
-        depth,
-        nav_depth
-    )
-
-    # list of transformers to apply, adding user-specific ones if user is provided
-    transformers = [blocks_api_transformer]
+    # create ordered list of transformers, adding BlocksAPITransformer at end.
+    transformers = BlockStructureTransformers()
     if user is not None:
         transformers += COURSE_BLOCK_ACCESS_TRANSFORMERS + [ProctoredExamTransformer()]
+    transformers += [
+        BlocksAPITransformer(
+            block_counts,
+            student_view_data,
+            depth,
+            nav_depth
+        )
+    ]
```

Comment:
Hmm.. yes.  Let's delay the decision on this.  I'm completely with you, though, on having a single list of transformers instead of different ones for super-user access.
Perhaps the transformers should declare themselves as access-control types.  Otherwise, they all special case Super-User access.  

By the way, anonymous access works since we would pass the `AnonymousUser` through.  In this particular case, for Insights, we just don't want to check access - i.e., treat as SuperUser access.